### ISSUE TITLE
Code Duplication across pages: Reduce

### ISSUE THREAD

**Issue by danielbeardsley** [iFixit/react-commerce#572]
### Problem
We now have [five pages](https://github.com/iFixit/react-commerce/tree/main/frontend/pages). They are 100-140 lines long and most of them are *nearly* identical. five copies of nearly identical code can be hard to maintain.

### Do This
Reduce duplication. Not sure if extracting re-used bits is better, or making a more generic *Page* component is better.

Here's a diff of two of the files:
![image](https://user-images.githubusercontent.com/26855/182500343-467268c4-8c96-486c-a0f3-8baf99b83d5c.png)

**Comment by dhmacs** [comment]
In the new product page branch I'm introducing a new `templates` folder to solve this code redundancy 👍 Might be worth to extrapolate into a separate PR for the product list page

**Comment by sterlinghirsh** [comment]
@dhmacs can we merge that branch or extract that relevant change so we can use the new pattern asap? We don't want to wait on the product page being done before we address this.
